CLOUD-349, CLOUD-14: Document setup of Dashboard GitHub Enterprise Integration (#4336)
cypress-io/cypress-documentation: content/guides/dashboard/github-integration.md

3. Complete the **Webhook** section.
   - Enter the **Webhook URL**, https://dashboard.cypress.io/webhooks/github-app
   - Copy and paste the provided **Webhook secret** from your Cypress Dashboard
     to **Webhook secret (optional)**.

<DocsImage src="/img/dashboard/github-integration/ghe/ghe-03.png" alt="Configure app webhook" id="cypress-github-enterprise-app-installation-process-3"></DocsImage>

4. Set the **Repository Permissions**. Below are the minimum permissions
   required for the new GitHub App.
   - **Actions**: read-only
   - **Checks**: read & write
   - **Contents**: read-only
   - **Discussions**: read & write
   - **Pull requests**: read & write
   - **Commit statuses**: read & write

<DocsImage src="/img/dashboard/github-integration/ghe/ghe-04.png" alt="Configure app permissions" id="cypress-github-enterprise-app-installation-process-4"></DocsImage>

5. Skip to the bottom of the form and click the **Create GitHub App** button.
   Your new GitHub App is now created, and you'll be taken to the settings page.

6. Returning to the Cypress Dashboard GitHub Enterprise Integration

>    configuration screen, populate the following values from your new GitHub App

Reviewer: ```suggestion
   configuration screen, populate the following values from your new GitHub App.
```

Author: Good catch!

Author: Resolved by https://github.com/cypress-io/cypress-documentation/pull/4336/commits/19bdd151fb95bca0b6b112d75de49fccf9f689c3#diff-031fb3f5fe6eb378cdd54e4a238398cf73ff1afa6543ae24741a3d2be8e85cd0R149

   - Copy the root URL of your GitHub Enterprise site to the Cypress **GitHub

>      Enterprise Server Url** field.

Reviewer 2: ```suggestion
     Enterprise Server URL** field.
```

Author: That's how the field is written in the Dashboard, are you sure you want to change it? We would need to change the dashboard to match.

Reviewer 2: Yes, it is being corrected as part of a separate Jira ticket to correctly reflect GitHub.

   - Copy the **App ID** to the Cypress **GitHub Enterprise App ID** field.
   - Copy the **Client ID** to the Cypress **GitHub Enterprise Client ID**
     field.

>    - Copy the **GitHub App name** to the Cypress **GitHub Enterprise app name**

Reviewer 2: ```suggestion
   - Copy the **GitHub App name** to the Cypress **GitHub Enterprise App name**
```

Author: That's how the field is written in the Dashboard, are you sure you want to change it? We would need to change the dashboard to match.

Reviewer 2: Yes, it is being corrected as part of a separate Jira ticket to correctly reflect GitHub.

>    - Copy the **GitHub App name** to the Cypress **GitHub Enterprise app name**

Reviewer 2: Still need to include a screenshot of the GitHub App name field from GitHub

Author: It's already there https://cypress-documentation-n6c1j1liv-cypress-io.vercel.app/guides/dashboard/github-integration#cypress-github-enterprise-app-installation-process-5

Reviewer 2: See it now. Was expecting the field, not the header, but that works. Will resolve.

<a id="cypress-github-enterprise-app-installation-process-4"></a>

4. Fill in the **Repository Permissions**.

   - **Actions**: read-only
   - **Checks**: read & write
   - **Contents**: read-only
   - **Discussions**: read & write
   - **Pull requests**: read & write
   - **Commit statuses**: read & write

<DocsImage src="/img/dashboard/github-integration/ghe/ghe-04.png" alt="Configure app permissions" ></DocsImage>

<a id="cypress-github-enterprise-app-installation-process-5"></a>

5. Skip to the bottom of the form and click the **Create GitHub App** button.
   You'll be taken to the app settings page for your new GitHub app. Copy and
   paste the new app details into the Dashboard.

   - Copy and paste the App name into the **GitHub Enterprise app name** field.
   - Copy and paste the App ID into the **GitHub Enterprise app ID** field.
   - Copy and paste the Client ID into the **GitHub Enterprise client ID**
     field.
   - Copy and paste the root URL of your GitHub Enterprise site into the
     **GitHub Enterprise Server Url** field.

<DocsImage src="/img/dashboard/github-integration/ghe/ghe-05.png" alt="Collect new app info" ></DocsImage>

> <a id="cypress-github-enterprise-app-installation-process-6"></a>

Reviewer 2: This screenshot should include the "GitHub App name" field with a highlight.

> 6. Click the **Generate a new client secret** button, then copy and paste the
>    generated secret into the **GitHub Enterprise client secret** field on the
>    Dashboard.

Reviewer 2: ```suggestion
```

<DocsImage src="/img/dashboard/github-integration/ghe/ghe-06.png" alt="Generate client secret" ></DocsImage>

<a id="cypress-github-enterprise-app-installation-process-7"></a>

> 7. Scroll to the bottom of the app settings page and click the **Generate a
>    private key** button. A file will be downloaded to your browser's default
>    downloads location. Open the file with a plain text editor, select all text
>    and copy and paste into the **GitHub Enterprise private key** field on the
>    Dashboard.

Reviewer 2: ```suggestion
```

<DocsImage src="/img/dashboard/github-integration/ghe/ghe-07.png" alt="Generate private key" ></DocsImage>

<a id="cypress-github-enterprise-app-installation-process-8"></a>

> 8. In the Dashboard, click the **Next Step** button and you will be taken to the

Reviewer 2: ```suggestion
7. In the Dashboard, click the **Next Step** button and you will be taken to the
```

   GitHub Enterprise app authorization page. Click the **Authorize \[your app
   name\]** button.

<DocsImage src="/img/dashboard/github-integration/ghe/ghe-08.png" alt="Authorize GitHub App" ></DocsImage>

<a id="cypress-github-enterprise-app-installation-process-9"></a>

> 9. On your newly-authorized GitHub App, click the **Install** button.

Reviewer 2: ```suggestion
8. On your newly-authorized GitHub App, click the **Install** button.
```

<DocsImage src="/img/dashboard/github-integration/ghe/ghe-09.png" alt="Install GitHub App" ></DocsImage>

<a id="cypress-github-enterprise-app-installation-process-10"></a>

> 10. Nearly there! On the GitHub App installation page, choose whether you want

Reviewer 2: ```suggestion
9. Nearly there! On the GitHub App installation page, choose whether you want
```

    to install the app against all repos or select specific ones, then click the
    **Install** button.

<DocsImage src="/img/dashboard/github-integration/ghe/ghe-10.png" alt="Confirm installation of GitHub App" ></DocsImage>

<a id="cypress-github-enterprise-app-installation-process-"></a>

> 11. Finally you will be returned to the Dashboard. Congratulations, you have
>     installed the Cypress Dashboard GitHub Enterprise integration! Treat
>     yourself to the refreshing beverage of your choice, then continue to the
>     next section.

Reviewer 2: ```suggestion
10. Finally, you will be returned to the Dashboard. Congratulations, you have
    installed the Cypress Dashboard GitHub Enterprise integration! You are now
    ready configure a GitHub Integration for a project.
```

3. Click the **Install** button to complete the installation.

### Cypress GitHub Enterprise app installation process

For GitHub Enterprise the installation is a little more involved, you need to
create a new GitHub App, copy various settings and credentials from the new app,
paste them into the Dashboard and go through the GitHub app activation process.

> 

Reviewer 2: Run on sentence. Suggest...
To install GitHub Enterprise Integration you need to; create a new GitHub App, copy the required settings and credentials from the new app to the Cypress configuration form, and complete the GitHub app activation process.

1. With your Dashboard organization's GitHub Enterprise page open, open a new
   tab or browser window and browse to your GitHub Enterprise site. Navigate to
   your GitHub organization -> **Settings** -> **Developer settings** ->
   **GitHub Apps**. Do **NOT** go to **OAuth Apps**. Click the **New GitHub
   App** button.

2. Fill in the **Register new GitHub app** form.

   - Enter a **GitHub App name**. Name may contain only dashes, letters and
     numbers, **no spaces**.
   - Enter the Cypress Dashboard **Homepage URL**, https://dashboard.cypress.io
   - Enter the **Callback URL** and **Setup URL**,
     https://dashboard.cypress.io/apps/github/callback

3. Fill in the **Webhook** form.

>    - Enter the **Webhook URL**, https://dashboard.cypress.io/webhooks/github-app

Reviewer 2: You can copy this directly from the Dashboard GitHub Enterprise configuration page.

   - Copy and paste your Cypress organization's **Webhook secret** from the

>      Dashboard into the GitHub Webhook form.

Reviewer 2: You can copy this directly from the Dashboard GitHub Enterprise configuration page.

1. Go to the project's settings page.
   <DocsImage src="/img/dashboard/visit-project-settings.png" alt="Visit project settings" ></DocsImage>

> 2. Scroll down to the GitHub Integration section.

Reviewer 2: ```suggestion
2. Scroll down to the GitHub Integration or GitHub Enterprise Integration section.
```
